spaceship/lib/spaceship/two_step_or_factor_client.rb

```diff
@@ -247,27 +275,29 @@ def request_two_factor_code_from_phone_choose(phone_numbers, code_length)
       available = phone_numbers.collect do |current|
         current['numberWithDialCode']
       end
-      chosen = choose(*available)
+      chosen = choose_phone_number(available)
       phone_id = phone_id_from_masked_number(phone_numbers, chosen)
 
       request_two_factor_code_from_phone(phone_id, chosen, code_length)
     end
 
     # this is used in two places: after choosing a phone number and when a phone number is set via ENV var
-    def request_two_factor_code_from_phone(phone_id, phone_number, code_length)
-      # Request code
-      r = request(:put) do |req|
-        req.url("https://idmsa.apple.com/appleauth/auth/verify/phone")
-        req.headers['Content-Type'] = 'application/json'
-        req.body = { "phoneNumber" => { "id" => phone_id }, "mode" => "sms" }.to_json
-        update_request_headers(req)
-      end
+    def request_two_factor_code_from_phone(phone_id, phone_number, code_length, should_request_code = true)
```

Comment:
should this maybe be renamed and the `should_request_code` part be called explicitly before calling this method in the places where `should_request_code = true`? might make it clearer how the code "flows"

Reply:
I gave the `should_request_code` a default value to keep the behavior of this method the same as prior to these changes unless `should_request_code` was explicitly provided, but given that this method isn't called anywhere else I agree we should make it more clear.

Reply:
@janpio Sorry, I may have misunderstood, can you clarify if you mean explicitly passing a value for `should_request_code` rather than having a default value, or if you mean separating out the call to request a code into its own method?

Reply:
I meant renaming this method to something without "request" and have the request part in an explicit method call before. But doesn't matter too much.

Reply:
I think this should wait for a refactor in a future PR, breaking up this method would affect other areas of the code and make the big conditional block even more unwieldy.